drivers/i2c/i2c_ll_stm32_v1.c

```diff
@@ -210,6 +210,7 @@ static inline void handle_addr(struct device *dev)
 		}
 	}
 
+#ifdef CONFIG_SOC_SERIES_STM32F1X
 	if (data->current.is_write || data->current.len > 2) {
```

Comment:
It seems that in F1 case you need to do a different process than in other cases. I would like to ask you
to describe in the commit message what is happening here from I2C protocol standpoint. Are you sure that
this different behaviour is only for F1? Maybe having a clear vision from I2C protocol
standpoint would help us to understand what is going on on other SoC series.

Reply:
Yep. In the F1 family there are some silicon limitations due to errata. As you may see there is a couple of corner cases for data count below 3. From the perspective of i2c proto there nothing to do. I's just a workaround recommended by the errata.

Reply:
> Yep. In the F1 family there are some silicon limitations due to errata. As you may see there is a couple of corner cases for data count below 3. From the perspective of i2c proto there nothing to do. I's just a workaround recommended by the errata.

Can you mention the errata link in the commit message? Or maybe in a comment in the code.

Reply:
Done. Link is too long https://www.st.com/content/ccc/resource/technical/document/errata_sheet/f5/50/c9/46/56/db/4a/f6/CD00197763.pdf/files/CD00197763.pdf/jcr:content/translations/en.CD00197763.pdf
So i just add a reference name of ducument.

Reply:
STM32F103 reference manual [RM0008](https://www.st.com/content/ccc/resource/technical/document/reference_manual/59/b9/ba/7f/11/af/43/d5/CD00171190.pdf/files/CD00171190.pdf/jcr:content/translations/en.CD00171190.pdf) and STM32F401 reference manual [RM0368](https://www.st.com/content/ccc/resource/technical/document/reference_manual/5d/b1/ef/b2/a1/66/40/80/DM00096844.pdf/files/DM00096844.pdf/jcr:content/translations/en.DM00096844.pdf) describe differently the master reception for N < 3.

F1: 
```
The cases where a single byte or two bytes are to be received should be handled differently, as described below:
Case of a single byte to be received:
- In the ADDR event, clear the ACK bit.
- Clear ADDR
- Program the STOP/START bit
- Read the data after the RxNE flag is set.

Case of two bytes to be received:
- Set POS and ACK
- Wait for the ADDR flag to be set
- Clear ADDR
- Clear ACK
- Wait for BTF to be set
- Program STOP
- Read DR twice
```

F4:
```
For 2-byte reception:
- Wait until ADDR = 1 (SCL stretched low until the ADDR flag is cleared)
- Set ACK low, set POS high
- Clear ADDR flag
- Wait until BTF = 1 (Data 1 in DR, Data2 in shift register, SCL stretched low until a data 1 is read)
- Set STOP high
- Read data 1 and 2
```
@pavlohamov Does the "else" part work for F1 family if we just handle the 0 length case:
```
if (data->current.is_write) {
	LL_I2C_ClearFlag_ADDR(i2c);
	return;
}
if (data->current.len == 0U) {
	LL_I2C_ClearFlag_ADDR(i2c);
} else if (data->current.len == 1U) {
	/* Single byte reception: enable NACK and clear POS */
	LL_I2C_AcknowledgeNextData(i2c, LL_I2C_NACK);
} else if (data->current.len == 2U) {
	/* 2-byte reception: enable NACK and set POS */
	LL_I2C_AcknowledgeNextData(i2c, LL_I2C_NACK);
	LL_I2C_EnableBitPOS(i2c);
}
LL_I2C_ClearFlag_ADDR(i2c);
```

Reply:
> Done. Link is too long https://www.st.com/content/ccc/resource/technical/document/errata_sheet/f5/50/c9/46/56/db/4a/f6/CD00197763.pdf/files/CD00197763.pdf/jcr:content/translations/en.CD00197763.pdf
> So i just add a reference name of document.

Yes, name is fine. But pls add also the section in the errata for the workaround.

Reply:
> @pavlohamov Does the "else" part work for F1 family if we just handle the 0 length case:

I see your point.
0 and 1 should be handled in the same way. I'm not able to find the case in the errata though.

Reply:
@ydamigos Unfortunate no. 
1) Handling of zero length breaks F4 procedure
2) Reading of 1 bytes lead to two actual bytes "read". At least clock is generated.
3) Reading of two bytes generates NACK after first byte, so slave send nothing. 

Code that is working properly seems to me a bit odd...
```
if (data->current.is_write) {
	LL_I2C_ClearFlag_ADDR(i2c);
	return;
}
if (data->current.len == 0U && IS_ENABLED(CONFIG_SOC_SERIES_STM32F1X)) {
	LL_I2C_GenerateStopCondition(i2c);
} else if (data->current.len == 1U) {
	/* Single byte reception: enable NACK and clear POS */
	LL_I2C_AcknowledgeNextData(i2c, LL_I2C_NACK);
#ifdef CONFIG_SOC_SERIES_STM32F1X
	LL_I2C_ClearFlag_ADDR(i2c);
	LL_I2C_GenerateStopCondition(i2c);
#endif
} else if (data->current.len == 2U) {
#ifdef CONFIG_SOC_SERIES_STM32F1X
	LL_I2C_ClearFlag_ADDR(i2c);
#endif
	/* 2-byte reception: enable NACK and set POS */
	LL_I2C_AcknowledgeNextData(i2c, LL_I2C_NACK);
	LL_I2C_EnableBitPOS(i2c);
}
LL_I2C_ClearFlag_ADDR(i2c);
```